src/entities/Task/model/slice/tasksSlice.ts

```diff
@@ -0,0 +1,87 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import {
+    StatusEnum, SubTask, Task, TasksSchema,
+} from '../types/task';
+
+const initialState: TasksSchema = {
+    data: [],
+};
+
+const updateLocalStorage = (tasks: Task[]) => {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const getTaskById = (tasks: (Task | SubTask)[], id: string) => tasks.find((task) => task.id === id);
+
+export const tasksSlice = createSlice({
+    name: 'tasks',
+    initialState,
+    reducers: {
+        getAllTasks: (state) => {
+            state.data = JSON.parse(localStorage.getItem('tasks') || '[]');
+        },
+
+        createNewTask: (state, action: PayloadAction<Task>) => {
+            state.data.push(action.payload);
+            updateLocalStorage(state.data);
+        },
+
+        changeTaskStatus: (state, action: PayloadAction<Task>) => {
```

Comment:
for change status functions there's no need to pass whole Task/SubTask in action